src/network.rs

```diff
@@ -78,40 +78,26 @@ fn test_parse_addrtype() {
 }
 
 pub fn parse_unicast_addr(addrtype: &SdpAddrType, value: &str) -> Result<IpAddr, SdpParserError> {
```

Comment:
There a few places in the code that (seem to) rely on this checking. Should we move the checking to another function or inline the checks? I would be for using a function so there is less duplicate code.